shared/editor/embeds/Grist.tsx

```diff
@@ -0,0 +1,17 @@
+import * as React from "react";
+import Frame from "../components/Frame";
+import { EmbedProps as Props } from ".";
+
+const URL_REGEX = new RegExp(
+  "^https?://.*/p/([0-9][0-9]*)\\?embed=true$" // Ensure its a page URL
```

Comment:
The regex seems to be too weak. I'm assuming it's kept this way in order to support the possibility of Grist being self hosted. But, did you also consider the possibility of this being interfering/colliding with other supported embeds? What do you think @tommoor ?

Reply:
Agreed that it's too loose, suggest we change to only support saas-hosted to make this mergable and then revisit how to support self-hosted.

A pattern is needed for this (and some other integrations) where there is a section under Settings -> Integrations that you can enter your self-hosted url which will then be used to calculate the match here.

Reply:
Yes, that is why. I forced the /p/1?embed=true at the end but thats as much as I could do. Any suggestions besides change it to only work on the SaaS product?

Reply:
> Any suggestions besides changing it to only work on the SaaS product?

I second @tommoor 's suggestion on this. Honestly, a weaker regex might find its place in a specific, closed setting as a quick hack but pushing it upstream is not a good idea owing to unforeseen, negative consequences.

Reply:
Code updated. It would be great to support self-hosted versions at some point. (that is one of the main reasons I support Outline over competing products).

Reply:
Yep, as mentioned above we need to prioritize the pattern as it'll benefit multiple integrations including draw.io/diagrams.net as well.